spec/models/reservation_spec.rb

```diff
@@ -1,725 +1,913 @@
+      set = Array.new(2) { ReservationMock.new(equipment_item_id: 1) }
+      expect(described_class.unique_equipment_items?(set)).to be_falsey
     end
-
-    it { is_expected.not_to be_valid }
-    it 'should not save' do
-      expect(reservation.save).to be_falsey
-      expect(Reservation.all.size).to eq(0)
-    end
-    it 'cannot be updated' do
-      reservation.start_date = Time.zone.today + 1.day
-      expect(reservation.save).to be_falsey
-    end
-    # it 'fails appropriate validations' do
-    #   reservation.should_not be_not_empty
-    #   Reservation.validate_set(reservation.reserver,
-    #                            [] << reservation).should_not == [] #fails
-    # end
-    # it 'passes other custom validations' do
-    #   reservation.should be_no_overdue_reservations
-    #   reservation.should be_start_date_before_due_date
-    #   reservation.should be_not_in_past
-    #   reservation.should be_matched_item_and_model
-    #   reservation.should be_duration_allowed # fails: tries to run
-    # validations on nil
-    #   reservation.should be_start_date_is_not_blackout
-    #   reservation.should be_due_date_is_not_blackout
-    #   reservation.should be_available #fails: tries to run validations on nil
-    #   reservation.should be_quantity_eq_model_allowed # fails: tries to run
-    # validations on nil
-    #   reservation.should be_quantity_cat_allowed # fails: tries to run
-    # validations on nil
-    # end
-    it 'updates with equipment model' do
-      reservation.equipment_model = FactoryGirl.create(:equipment_model)
-      FactoryGirl.create(:equipment_item,
-                         equipment_model: reservation.equipment_model)
-      expect(reservation.save).to be_truthy
-      expect(reservation).to be_valid
-      expect(Reservation.all.size).to eq(1)
+    it 'returns true when the set has no duplicate items' do
+      set = Array.new(2) { |i| ReservationMock.new(equipment_item_id: i) }
+      expect(described_class.unique_equipment_items?(set)).to be_truthy
     end
   end
 
-  context 'editing the due date' do
-    context 'while not checked out' do
-      shared_examples_for 'does not affect the overdue status' do |res_type|
-        subject(:reservation) do
-          r = FactoryGirl.build(res_type)
-          r.save(validate: false)
-          r
-        end
-
-        it 'regardless of date' do
-          reservation.due_date = Time.zone.today - 1.day
-          expect { reservation.save }.not_to change { reservation.overdue }
-          reservation.due_date = Time.zone.today + 1.day
-          expect { reservation.save }.not_to change { reservation.overdue }
-        end
+  describe 'custom validations' do
+    it 'is invalid when start date after due date' do
+      res = FactoryGirl.build(:valid_reservation,
+                              start_date: Time.zone.today + 3.days,
+                              due_date: Time.zone.today)
+      expect(res.start_date_before_due_date).not_to be_empty
```

Comment:
Using a shared example would complicate the setup step for most of the tests by requiring a separate `context` with `let!` blocks because most of these require other models to have certain states for them to fail. Also, the way it is now allows you to see the validation that's failing right away; this could be done in a shared example but we would lose the information about the kind of data that is supposed to make it fail.